src/Agent.Sdk/LogPlugin.cs

```diff
@@ -0,0 +1,534 @@
+                                      }))
+            {
+                while (!_shortCircuited[typeName].Task.IsCompleted && !token.IsCancellationRequested)
+                {
+                    while (!_shortCircuited[typeName].Task.IsCompleted && _outputQueue[typeName].TryDequeue(out string line))
+                    {
+                        try
+                        {
+                            var id = line.Substring(0, line.IndexOf(":"));
+                            var message = line.Substring(line.IndexOf(":") + 1);
+                            var processLineTask = plugin.ProcessLineAsync(context, _steps[id], message);
+                            var completedTask = await Task.WhenAny(_shortCircuited[typeName].Task, processLineTask);
+                            if (completedTask == processLineTask)
+                            {
+                                await processLineTask;
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            // ignore exception
+                            // only trace the first 10 errors.
+                            if (errors.Count < 10)
+                            {
+                                errors.Add(ex.ToString());
+                            }
+                        }
+                    }
+
+                    // back-off before pull output queue again.
+                    await Task.Delay(500);
+                }
+            }
+
+            // process all remaining outputs
+            context.Trace("Process remaining outputs after job finished.");
+            while (!_shortCircuited[typeName].Task.IsCompleted && _outputQueue[typeName].TryDequeue(out string line))
+            {
+                try
+                {
+                    var id = line.Substring(0, line.IndexOf(":"));
+                    var message = line.Substring(line.IndexOf(":") + 1);
+                    var processLineTask = plugin.ProcessLineAsync(context, _steps[id], message);
+                    var completedTask = await Task.WhenAny(_shortCircuited[typeName].Task, processLineTask);
+                    if (completedTask == processLineTask)
+                    {
+                        await processLineTask;
+                    }
+                }
+                catch (Exception ex)
+                {
+                    // ignore exception
+                    // only trace the first 10 errors.
+                    if (errors.Count < 10)
+                    {
+                        errors.Add(ex.ToString());
+                    }
+                }
+            }
+
+            // print out the plugin has been short circuited.
+            if (_shortCircuited[typeName].Task.IsCompleted)
+            {
+                if (initialized)
+                {
+                    context.Output($"Plugin has been short circuited due to exceed memory usage limit.");
```

Comment:
Include plugin name.

Reply:
sure

Reply:
actually, i don't have to, since the `context.Output()` will prefix everything with `context.Name`